src/main.rs

```diff
@@ -41,44 +35,105 @@ fn main() -> Result<()> {
+
+                    match cmd.subcommand() {
+
+                        (res, Some(command)) => {
+                            let data = util::json_parse(command.value_of(Parameters::data))?;
+                            let id = command.value_of(Parameters::id).unwrap();
+                            
+                            let nam = Resources::from_str(res);
+                
+                            match nam? {
+                                Resources::app => apps::create(&url, id, data)?,
+                                Resources::device => {
+                                    let app_id = command.value_of(Resources::app).unwrap();
+                                    devices::create(&url, id, data, app_id)?
+                                },
+                            }
+                        }
+                        _ => println!("{}", NO_COMMAND_MESSAGE)
+
+                    }
+
+                }
+                Verbs::delete => {
+
+                    match cmd.subcommand() {
+
+                        (res, Some(command)) => {
+                            let id = command.value_of(Parameters::id).unwrap();
+                            let nam = Resources::from_str(res);
+                
+                            match nam? {
+                                Resources::app => apps::delete(&url, id)?,
+                                Resources::device => {
+                                    let app_id = command.value_of(Resources::app).unwrap();
+                                    devices::delete(&url, app_id, id)?
+                                },
+                            }
+                        }
+                        _ => println!("{}", NO_COMMAND_MESSAGE)
+
+                    }
+
+                }
+                Verbs::edit => {
+
+                    match cmd.subcommand() {
+
+                        (res, Some(command)) => {
+                            let id = command.value_of(Parameters::id).unwrap();
+                            
+                            let nam = Resources::from_str(res);
+                
+                            match nam? {
+                                Resources::app => apps::edit(&url, id),
+                                Resources::device => {
+                                    let app_id = command.value_of(Resources::app).unwrap();
+                                    devices::edit(&url, app_id, id)
+                                },
+                            }
+                        }
+                        _ => println!("{}", NO_COMMAND_MESSAGE)
+
+                    }
+
+
+                }
+                Verbs::get => {
+
+                    match cmd.subcommand() {
+
+                        (res, Some(command)) => {
+                            let id = command.value_of(Parameters::id).unwrap();
+                            
+                            let nam = Resources::from_str(res);
```

Comment:
Please use another variable name. `name` and `nam` is confusing. I suggest `verb` and `resource`

Reply:
I apologize for this.